### PR TITLE
Add parter auth flow

### PR DESCRIPTION
**Pull request by jinalvirani**

### CODE REVIEW

File: src/commands/auth.ts [hunk 1]
```diff
@@ -5,53 +5,53 @@ import CommandError from '../lib/CommandError';
...
-        throw new CommandError('Enter a valid mobile number');
-      }
-      await Auth.loginInWithMobile(mobile);
-    }
-    else command.help()
-  } catch (error) {
-    throw new CommandError(error.message, error?.code);
-  }
-};
+// const AuthenticationHandler = async (options, command) => {
```
Comment (vivek8690) [inline src/commands/auth.ts]: Remove unused code @jinalvirani

File: src/helper/constants.ts [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
-export const THEME_COMMANDS = [ 'publish', 'pull', 'pull-config', 'serve', 'sync', 'unpublish'];
+export const THEME_COMMANDS = ['publish', 'pull', 'pull-config', 'serve', 'sync', 'unpublish'];
 export const ENVIRONMENT_COMMANDS = ['env'];
 export const AUTHENTICATION_COMMANDS = ['auth', 'login', 'logout'];
 export const EXTENSION_COMMANDS = ['init', 'setup', 'get', 'set'];
-export const PARTNER_COMMANDS = ['connect']
\ No newline at end of file
+export const PARTNER_COMMANDS = ['connect']
+
+export const ALLOWD_ENV = [
```
Comment (vivek8690) [inline src/helper/constants.ts]: For now allow for all the domains @jinalvirani

**Review by vivek8690 — CHANGES_REQUESTED** [review]

File: src/lib/Theme.ts [hunk 3]
```diff
@@ -1112,8 +1112,37 @@ export default class Theme {
...
+        const config = fs.readJSONSync(configPath);
+        config.theme.available_sections = available_sections
+        config.theme.assets = {};
+        config.theme.assets.umd_js = {};        
+        config.theme.assets.umd_js.links = umdJsUrls;
+        config.theme.assets.common_js = {};        
+        config.theme.assets.common_js.link = commonJsUrl;
+        config.theme.assets.css = {};   
+        config.theme.assets.css.links = cssUrls;
+        await fs.writeJson(configPath, config,{ spaces: 2 });
```
Comment (Shivraj0) [inline src/lib/Theme.ts]: This will affect places where `config.json` is read, check once.

File: src/lib/Theme.ts [hunk 4]
```diff
@@ -1112,8 +1112,37 @@ export default class Theme {
...
         }
     };
+    
+    public static generateAssets = async () =>{
+        let available_sections = await Theme.getAvailableSectionsForSync();
+        await Theme.validateAvailableSections(available_sections);
+        Theme.clearPreviousBuild();
+        const imageCdnUrl = await Theme.getImageCdnBaseUrl();
+        const assetCdnUrl = await Theme.getAssetCdnBaseUrl();
+        Theme.createVueConfig();
```
Comment (Shivraj0) [inline src/lib/Theme.ts:2071]: Not required, remove this.

File: src/lib/Theme.ts [hunk 5]
```diff
@@ -1112,8 +1112,37 @@ export default class Theme {
             throw new CommandError(err.message, err.code);
         }
     };
+    
+    public static generateAssets = async () =>{
+        let available_sections = await Theme.getAvailableSectionsForSync();
+        await Theme.validateAvailableSections(available_sections);
+        Theme.clearPreviousBuild();
```
Comment (Shivraj0) [inline src/lib/Theme.ts]: Move this on first line in `generateAssets` function.

File: src/lib/Theme.ts [hunk 6]
```diff
@@ -1112,8 +1112,37 @@ export default class Theme {
             throw new CommandError(err.message, err.code);
         }
     };
+    
+    public static generateAssets = async () =>{
```
Comment (Shivraj0) [inline src/lib/Theme.ts]: Add comment here so as to why this function is required.

**Review by Shivraj0 — COMMENTED** [review]
Reviewed commit. Check if we can fix the hidden files issue which also gets archived in `fdk theme package` command.

File: src/lib/Theme.ts (hunk 5 above)
Comment (jinalvirani) [reply src/lib/Theme.ts]: Already added inside theme package cmd before calling generateAssets

File: src/lib/Theme.ts (hunk 3 above)
Comment (jinalvirani) [reply src/lib/Theme.ts]: Done

**Pull request merged**